cmd/acra-connector/acra-connector.go

```diff
@@ -229,6 +229,14 @@ func main() {
 	tlsCert := flag.String("tls_cert", "", "Path to certificate")
 	tlsAcraserverSNI := flag.String("tls_acraserver_sni", "", "Expected Server Name (SNI) from AcraServer")
 	tlsAuthType := flag.Int("tls_auth", int(tls.RequireAndVerifyClientCert), "Set authentication mode that will be used in TLS connection with AcraServer/AcraTranslator. Values in range 0-4 that set auth type (https://golang.org/pkg/crypto/tls/#ClientAuthType). Default is tls.RequireAndVerifyClientCert")
+	tlsOcspURL := flag.String("tls_ocsp_url", "", "OCSP service URL")
+	tlsOcspRequired := flag.String("tls_ocsp_required", "denyUnknown", "Whether we need OCSP response in order to accept certificate")
+	tlsOcspFromCert := flag.String("tls_ocsp_from_cert", "prefer", "How should we treat OCSP server described in certificate itself")
+	tlsOcspCheckWholeChain := flag.Bool("tls_ocsp_check_whole_chain", false, "Whether to check whole certificate chain, false = only end certificate")
+	tlsCrlURL := flag.String("tls_crl_url", "", "CRL URL")
+	tlsCrlFromCert := flag.String("tls_crl_from_cert", "use", "How should we treat CRL URL described in certificate itself")
+	tlsCrlCacheSize := flag.Int("tls_crl_cache_size", 16, "Size of in-memory LRU cache for storing fetched CRLs, 0 = unlimited")
+	tlsCrlCacheTime := flag.Int("tls_crl_cache_time", 60, "How long cached CRL is considerend valid, seconds, max = 300")
```

Comment:
Making coordinated changes between multiple binaries may be bothersome, so it would be nice to pack all these command-line parameters into some shared place.

An approach that I have been using at times is to declare some
```go
type OCSPParameters struct {
        URL string
        // ...
}
```
which has a `Register()` method to register the command-line parameters, then have a global one somewhere.

This approach allows for command-line reuse *and* you can pass a single parameter struct to `network.NewOCSPConfig()` constructor instead of adding more and more parameters.

Or it might be possible to reuse existing `OCSPConfig` struct for that...

Reply:
Sounds like a good idea, but one issue is that there are two URIs: for client-side and for DB side. And when parsing CLI args we cannot know which one we need in this specific `OCSPConfig` because there'll be two of them, each for different `TLSConfig`.